File: ats-ness-front-prod/src/app/demo/models/BesoinsTechnologies.ts
import {Candidat} from "./condidat";
import {Besoin} from "./besoin";

export class BesoinsTechnologies {
    id: number | undefined;

    technologie: string | undefined;
    niveau: string | undefined;
    importance: string | undefined;
    anneesExperience:number | undefined;

    // Relation avec l'entité Candidat
    besoin: Besoin | undefined;

}

File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/BesoinsTechnologies.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder

public class BesoinsTechnologies  {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String technologie;
    private String niveau;
    private String importance;
    private int anneesExperience;




    @JsonIgnore
    @ManyToOne
    private Besoins besoin;

}
